QGCCommon.pri

```diff
@@ -90,7 +90,6 @@ linux {
         QMAKE_CXXFLAGS -= -Zc:strictStrings
         QMAKE_CXXFLAGS_RELEASE -= -Zc:strictStrings
         QMAKE_CXXFLAGS_RELEASE_WITH_DEBUGINFO -= -Zc:strictStrings
-        QMAKE_CXXFLAGS += /std:c++17
```

Comment:
Why is this being removed? And why is it ok to remove it?

Reply:
c++17 standard was only required to eliminate a couple of warnings, but it also made the windows build not respect the version set in qgroundcontrol.pro (and in fact throw errors if the version there was changed).

Reply:
> but it also made the windows build not respect the c++ standard set in qgroundcontrol.pro

Why is the C++ version also set in qgroundcontrol.pro? That seems wrong. Since only QGCommon.pri is supposed to be used by all source as the settings for compiler stuff. If there is also C++ version flags in qgroundcontrol.pro shouldn't they be removed from there? And if QGCommon.pri needs to be updated to some other version, then update it there.

Reply:
I don't understand the historical reason why the standard was set in the project file, but I moved it to QGCCommon.pri now

Reply:
Which then comes back to why is there a need to drop down from c++17 to c++14? What is the reason? Moving backwards seems wrong.

Reply:
as far as reason, when changing the c++ standard on other builds (linux, eg), the way the c++ standard for windows build was configured (via QMAKE flags instead of via CONFIG) caused two c++ standards to be specified on the windows build...which was the cause of the original build failure.

So my changes around the c++ standard were to standardize how the c++ standard was specified.

Reply:
Sorry didn't realize windows build only. More strangeness. Would be good to see if c++17 works fine for all builds.

Reply:
All of the checks passed with this configuration. I did run QGC with this configuration on Ubuntu and didn't notice any issues.